txp/Pos/Txp/Logic/Local.hs

```diff
@@ -62,31 +67,39 @@ txProcessTransaction itw@(txId, txAux) = do
                    toList $
                    NE.zipWith (liftM2 (,) . Just) _txInputs resolvedOuts
     pRes <- modifyTxpLocalData $
-            processTxDo resolved toilEnv tipDB itw
+            processTxDo resolved toilEnv tipDB itw bootEra
     case pRes of
         Left er -> do
             logDebug $ sformat ("Transaction processing failed: "%build) txId
             throwError er
         Right _   ->
             logDebug (sformat ("Transaction is processed successfully: "%build) txId)
   where
+    notBootRelated =
+        let txDistr = getTxDistribution $ taDistribution txAux
+            inBoot (s,_) = s `HS.member` genesisBootProdStakeholders
+        in NE.filter (\txOutDistr -> any (not . inBoot) txOutDistr) txDistr
+
     processTxDo
         :: Utxo
         -> ToilEnv
         -> HeaderHash
         -> (TxId, TxAux)
+        -> Bool
         -> TxpLocalDataPure
         -> (Either ToilVerFailure (), TxpLocalDataPure)
-    processTxDo resolved toilEnv tipDB tx txld@(uv, mp, undo, tip, ())
-        | tipDB /= tip = (Left $ ToilTipsMismatch tipDB tip, txld)
+    processTxDo resolved toilEnv tipDB tx bootEra txld@(uv, mp, undo, tip, ())
+        | tipDB /= tip =
+            (Left $ ToilTipsMismatch tipDB tip, txld)
+        | not isDevelopment && bootEra && not (null notBootRelated) =
```

Comment:
No, I mean, you shouldn't use `genesisBootProdStakeholders` in development mode.